src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs

```diff
@@ -90,6 +90,7 @@ internal override async Task<AuthenticationResult> ExecuteAsync(CancellationToke
 
         private async Task AcquireAuthorizationAsync(CancellationToken cancellationToken)
         {
+            AuthenticationRequestParameters.RedirectUri = _webUi?.ValidateRedirectUri(AuthenticationRequestParameters.RedirectUri);
```

Comment:
This is the second part where I am a bit unsure about the implementation, because for all intents and purposes AuthenticationRequestParameters is immutable. 

However the redirect Uri is used in 2 places: 

- to get a code via the webUI
- to exchange the code for a token

Semantically, the new RedirectUri is the one we use everywhere, so it makes sense to update the value on AuthenticationRequestParameters. But it breaks immutability.

@MarkZuber